File: warehouse/UserSystem.java
package com.warehouse;

public class UserSystem {  
    private String name;
    private String role;
    private String password;

    public UserSystem() {}; 
    
    public UserSystem(String name, String role, String password) {
        this.name = name;
        this.role = role;
        this.password = password;
    }
    

    public boolean login(String role, String password) {
        return this.role.equals(role) &&  this.password.equals(password);
    }
    
    public boolean logout() {
        System.out.println("User " + name + " logged out");
        return true;
    }
    
    public String getName(){
        return this.name;
    }
    
    public String getRole(){
        return this.role;
    }
}

File: src/UserSystem.cpp
// Implementation of UserSystem
#include "UserSystem.hpp"
#include <iostream>
using namespace std;

UserSystem::UserSystem() : name(""), password(""), role("") {}

UserSystem::UserSystem(const string& name, const string& password, const string& role) : name(name), password(password), role(role) {}

UserSystem::~UserSystem() {}

bool UserSystem::login(const string& name, const string& password, const string& role) {
    if (this->name == name && this->password == password && this->role == role) {
        cout << "Авторизация успешна: " << name << ", роль: " << role << endl;
        return true;
    }
    return false;
}

bool UserSystem::logout() {
    cout << "Пользователь вышел: " << name << endl;
    return true;
}

string UserSystem::getName() const {
    return name;
}

string UserSystem::getRole() const {
    return role;
}
